### PR TITLE
feat(signup oAuth): still need someone to test fb, otherwise working

### PR DESCRIPTION
**Pull request by aperez25**
need someone to test fb signup

### CODE REVIEW

File: app/components/Signup.test.jsx [hunk 1]
```diff
@@ -1,64 +1,64 @@
...
+    })
 
-//     // it('calls preventDefault', () => {
-//     //   expect(submitEvent.preventDefault).to.have.been.called
-//     // })
-//   })
-// })
+    // it('calls preventDefault', () => {
+    //   expect(submitEvent.preventDefault).to.have.been.called
+    // })
```
Comment (glebec) [inline app/components/Signup.test.jsx]: Any reason this test is still commented-out?

File: app/components/Signup.jsx [hunk 2]
```diff
@@ -2,7 +2,10 @@ import React from 'react'
 import { Link } from 'react-router'
 
 export const Signup = ({ signup }) => (
-  <div>
+  <form onSubmit={evt => {
+    evt.preventDefault()
+    signup(evt.target.name.value, evt.target.email.value, evt.target.password.value)
```
Comment (glebec) [inline app/components/Signup.jsx:7]: Sometimes with long function params I'll use carriage returns and indenting:

```js
signup(
  evt.target.name.value,
  evt.target.email.value,
  evt.target.password.value
)
```

File: server/auth.js [hunk 3]
```diff
@@ -112,14 +112,18 @@ auth.post('/signup/local', (req, res, next) =>
...
       email: req.body.email
-    }
+    },
+    attributes: {include: ['password_digest']}
   })
   .then(user => {
-    if (!user) User.create(req.body)
-    return user
+    if (!user) return User.create(req.body)
+    else return user.authenticate(req.body.password)
```
Comment (glebec) [inline server/auth.js:120]: For fun, an alternate format:

```js
.then(user => user
  ? User.create(req.body)
  : user.authenticate(req.body.password)
)
```

No need to actually use this format, just showing alternatives.

**Review by DillGromble — COMMENTED** [review]
merged master

**Review by DillGromble — APPROVED** [review]
merge master

**Pull request merged**